src/jit/lowerxarch.cpp

```diff
@@ -239,16 +239,17 @@ void Lowering::TreeNodeInfoInit(GenTree* tree)
 #if !defined(_TARGET_64BIT_)
 
         case GT_LONG:
-            if (tree->gtNext == nullptr)
+            if ((tree->gtLIRFlags & LIR::Flags::IsUnusedValue) != 0)
```

Comment:
I see even GenTree::isContained() is asserting that a contained node has a parent (the default case).  This assert condition too needs to be modified?

Source search of 'gtNext == nullptr', indicate similar condition in codegenarm64.cpp/codegenxarch.cpp::genCallInstruction() towards the end of the method

```
    // If there is nothing next, that means the result is thrown away, so this value is not live.
    // However, for minopts or debuggable code, we keep it live to support managed return value debugging.
    if ((call->gtNext == nullptr) && !compiler->opts.MinOpts() && !compiler->opts.compDbgCode)
    {
        gcInfo.gcMarkRegSetNpt(RBM_INTRET);
    }
```

In codegencommon.cpp::genFnEpilog() of arm64/amd64, the following assert seems to be correct to me.  You might want to double check:

```
        // amd64
        // If jmpNode is GT_JMP then gtNext must be null.
        // If jmpNode is a fast tail call, gtNext need not be null since it could have embedded stmts.
        noway_assert((jmpNode->gtOper != GT_JMP) || (jmpNode->gtNext == nullptr));

```

Feel free to address them in a separate PR.